powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CallbackUrlBehavior.java

```diff
@@ -169,7 +171,7 @@ public UpdateCallbackUrlResponse updateCallbackUrl(UpdateCallbackUrlRequest requ
                 throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_URL_FORMAT);
             }
 
-            callbackUrlRestClientManager.evictRestClientFromCache(entity);
+            restClientCache.refresh(entity.getId());
```

Comment:
🤔 Theoretically, the cache is not transactional, right? So if `#safe` throws an exception, the cache has been already invalidated. What about refreshing it at least after `#safe`?